back/boxtribute_server/business_logic/statistics/crud.py

```diff
@@ -206,3 +215,55 @@ def compute_top_products_donated(base_id):
 
     dimensions = _generate_dimensions("category", "product", "size", facts=facts)
     return {"facts": facts, "dimensions": dimensions}
+
+
+TARGET_ID_SEPARATOR = "---"
+
+
+def compute_moved_boxes(base_id):
+    """Count all boxes moved to locations in the given base, grouped by date of
+    movement, product category, and box state.
+    """
+    # This selects only information of boxes that were moved from InStock to Donated
+    # state, and are now in the base of given base ID. It is NOT taken into account that
+    # boxes can be moved back from Donated to InStock, nor that the product or other
+    # attributes of the box change after having been donated
+    selection = (
+        DbChangeHistory.select(
+            fn.MAX(DbChangeHistory.change_date).alias("moved_on"),
+            fn.CONCAT(Base.name, TARGET_ID_SEPARATOR, Location.name).alias("target_id"),
```

Comment:
Why do you add the Base name for the target_id? I would just do the Location.name for Outgoing Locations and the Base.name for shipments.

Reply:
makes sense :+1: